compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -200,7 +200,7 @@ struct MatcherPos<'root, 'tt> {
 
 // This type is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-rustc_data_structures::static_assert_size!(MatcherPos<'_, '_>, 192);
+rustc_data_structures::static_assert_size!(MatcherPos<'_, '_>, 240);
```

Comment:
Ouch, this may hurt compile times for crates that do a lot of macro expansion.

Reply:
https://github.com/rust-lang/rust/pull/94824#issuecomment-1064734294

Summertime sadness

Reply:
The additional size of `MatcherPos` is a consequence of this approved feature but I am open to alternatives

:shipit: Maybe `box` the new variant or the entire enum? IDK... :shipit:

Reply:
`MatcherPos` is cloned a lot, so more boxing won't help. I have a hacky optimization that avoids some of the cloning, I may make a PR for that next week.

Reply:
Good news! #95159 will reduce the size of this type from 240 bytes to 136 bytes.